src/utils/registryExperience.ts

```diff
@@ -3,22 +3,105 @@
-import * as vscode from 'vscode';
+import { AzureWizardPromptStep, ContextValueFilterQuickPickOptions, GenericQuickPickStep, IActionContext, IAzureQuickPickItem, QuickPickWizardContext, RecursiveQuickPickStep, runQuickPickWizard } from '@microsoft/vscode-azext-utils';
+import { CommonRegistryRoot, RegistryItem } from '@microsoft/vscode-docker-registries';
+import { TreeItem, TreeItemLabel, l10n } from 'vscode';
+import { UnifiedRegistryItem, UnifiedRegistryTreeDataProvider } from '../tree/registries/UnifiedRegistryTreeDataProvider';
 
-export async function registryExperience<TPick>(context: IActionContext, tdp: vscode.TreeDataProvider<unknown>, contextValueFilter: ContextValueFilter, skipIfOne: boolean = true): Promise<TPick> {
+export interface RegistryFilter {
+    /**
+     * This filter will exclude registry labels that you don't want to show in the quick pick.
+     */
+    exclude?: string[];
+}
+
+export interface RegistryExperienceOptions extends ContextValueFilterQuickPickOptions {
+    // if registryFilter is undefined, all registries will be shown in the quick pick
+    registryFilter?: RegistryFilter;
+}
+
+export async function registryExperience(context: IActionContext, tdp: UnifiedRegistryTreeDataProvider, options: RegistryExperienceOptions): Promise<UnifiedRegistryItem<RegistryItem>> {
+    // get the registry provider unified item
+    const registryProviderStep: AzureWizardPromptStep<QuickPickWizardContext>[] = [
+        new RegistryQuickPickStep(
+            tdp,
+            options
+        )
+    ];
+    const unifiedRegistryItem = await runQuickPickWizard(context, {
+        hideStepCount: true,
+        promptSteps: registryProviderStep,
+    });
+
+    // if no registries are found, throw an error
+    if (!unifiedRegistryItem) {
+        context.errorHandling.suppressReportIssue = true;
+        throw new Error(l10n.t('No registries found'));
+    }
+
+    // actually get the registry item from the provider
     const promptSteps: AzureWizardPromptStep<QuickPickWizardContext>[] = [
         new RecursiveQuickPickStep(
-            tdp,
+            (unifiedRegistryItem as UnifiedRegistryItem<RegistryItem>).provider,
             {
-                contextValueFilter: contextValueFilter,
-                skipIfOne: skipIfOne
+                ...options,
+                skipIfOne: true,
             }
         )
     ];
-
-    return await runQuickPickWizard(context, {
+    const pickWithoutProvider = await runQuickPickWizard(context, {
         hideStepCount: true,
         promptSteps: promptSteps,
     });
+
+    // add the provider back to the registry item
+    const pickWithProvider: UnifiedRegistryItem<RegistryItem> = {
+        wrappedItem: pickWithoutProvider,
+        provider: (unifiedRegistryItem as UnifiedRegistryItem<RegistryItem>).provider,
+        parent: undefined, // TODO: figure out how to get the parent
```

Comment:
Not exactly sure how to do this. An idea is we could add a method that takes an wrappedItem and the provider and returns the parent of the this wrappedItem. (It creates a new `UnifiedRegistryItem` by keep setting the parent field until it reaches wrappedItem.parent is undefined)

Reply:
logic similar to this for in a while loop: 

```TS
  return {
      provider: element.provider,
      wrappedItem: e,
      parent: element
  };
```
inspiration:https://github.com/microsoft/vscode-docker/blob/d3da812b31860dc3b9606a92a7e7717d7341efe8/src/tree/registries/UnifiedRegistryTreeDataProvider.ts#L41

Reply:
I think it would be better to run this as one single quick pick wizard instead of two separate ones. That would also solve this issue because you'd be picking the URTDP item instead of the wrapped BDP item.